src/render/canvas/canvas-renderer.ts

```diff
@@ -155,7 +155,9 @@ export class CanvasRenderer {
         const fontVariant = styles.fontVariant
             .filter(variant => variant === 'normal' || variant === 'small-caps')
             .join('');
-        const fontFamily = styles.fontFamily.join(', ');
+        const fontFamily = styles.fontFamily.map(fontName => {
+            return fontName.indexOf(' ') === -1 ? fontName : `"${fontName}"`;
```

Comment:
@grahamkane Is there a need to perform the `indexOf` test, or could we simplify this code by applying quoting to each map entry?

```javascript
const fontFamily = styles.fontFamily.map(fontName => `'${fontName}'`);
```

Reply:
FWIW - the W3 has some good [reference docs for the `font-family` property](https://www.w3.org/TR/CSS21/fonts.html#font-family-prop) - they list a number of invalid properties, and note that quoting is required for any font names that may also be keywords.

Given that, it may be easiest to apply quoting to all values.

Reply:
Edit: the suggested code snippet should be:

```javascript
const fontFamily = styles.fontFamily.map(fontName => `'${fontName}'`).join(', ');
```

(the `join` operation was previously missing)

Reply:
From [CSS Fonts Module Level 3](https://www.w3.org/TR/css-fonts-3/#family-name-value) (September, 2018):

> Font family names other than generic families must either be given quoted as strings, or unquoted as a sequence of one or more identifiers. This means most punctuation characters and digits at the start of each token must be escaped in unquoted font family names. 

So generic families do _not_ require quoting; my mistake earlier.

Reply:
Would it be possible to use the [isIdentToken](https://github.com/niklasvh/html2canvas/blob/4dd4a69c7a2e46c54b5d223fbc94fa4066f19a3c/src/css/syntax/parser.ts#L146) function to identify built-in font names instead?

It's possible this fix should be pushed down to the [`font-family` CSS property parser](https://github.com/niklasvh/html2canvas/blob/4dd4a69c7a2e46c54b5d223fbc94fa4066f19a3c/src/css/property-descriptors/font-family.ts#L14-L16).

I'm seeing slightly different behaviour between Chrome and Firefox with the fix proposed here in #1948 ; in particular I find that performing `styles.fontFamily.join(' ').split(',') ...` helps a little, since this joins all existing font-family tokens and _then_ splits them into individual font names.  I don't yet have a perfect solution though...

Reply:
The file that we are changing is placed inside node_modules in my project. How to fix then?

Reply:
Long story; in the end this train of thought led me to investigate the [`font-family` CSS property parser](https://github.com/niklasvh/html2canvas/blob/4dd4a69c7a2e46c54b5d223fbc94fa4066f19a3c/src/css/property-descriptors/font-family.ts#L14-L16) more thoroughly, and the bug is due to the fact that `Font Awesome 5 Free` (four words) parses to the family list `["Font", "Awesome", "5", "Free"]` (i.e. four non-existent font names).

A parser-level fix is provided in #2219.